### PR TITLE
split axum_http_server_factory in multiple files

### PR DESCRIPTION
**Pull request by bnjjj**
I splitted the `axum_http_server_factory.rs` file into several small ones.
also part of #1482

### CODE REVIEW

File: apollo-router/src/axum_factory/axum_http_server_factory.rs [hunk 1]
```diff
@@ -1413,7 +742,7 @@ mod tests {
 
         // Decompress body
         let body_bytes = response.bytes().await.unwrap();
-        let mut decoder = GzipDecoder::new(Vec::new());
+        let mut decoder = GzipEncoder::new(Vec::new());
```
Comment (Geal) [inline apollo-router/src/axum_factory/axum_http_server_factory.rs:745]: why replace decoder with encoder?

File: apollo-router/src/axum_factory/axum_http_server_factory.rs [hunk 2]
```diff
@@ -2727,42 +2056,6 @@ Content-Type: application/json\r
...
-        default_headers.append(VARY, HeaderValue::from_static("two"));
-        process_vary_header(&mut default_headers);
-        let vary = default_headers.get_all(VARY);
-        assert_eq!(vary.iter().count(), 2);
-        for value in vary {
-            assert!(value == "one" || value == "two");
-        }
-    }
-
     #[tokio::test]
```
Comment (Geal) [inline apollo-router/src/axum_factory/axum_http_server_factory.rs:2059]: could you move all the tests to a different file?

**Review by Geal — CHANGES_REQUESTED** [review]

File: apollo-router/src/axum_factory/axum_http_server_factory.rs (hunk 1 above)
Comment (o0Ignition0o) [reply apollo-router/src/axum_factory/axum_http_server_factory.rs:745]: https://app.circleci.com/pipelines/github/apollographql/router/6788/workflows/a18c2ea4-ec8a-4456-a181-c6a6a3ef815f/jobs/51621 didnt work too well :D

Comment (bnjjj) [reply apollo-router/src/axum_factory/axum_http_server_factory.rs:745]: Oops, a bad change :p

**Review by o0Ignition0o — APPROVED** [review]
LGTM! As a followup we can move the healthcheck endpoint and structs into its own separate file (which will grow over time)

File: apollo-router/src/lib.rs [hunk 3]
```diff
@@ -47,7 +47,7 @@ mod json_ext;
 #[macro_use]
 pub mod plugin;
 
-mod axum_http_server_factory;
```
Comment (Geal) [inline apollo-router/src/lib.rs:50]: did you forget to delete `src/axum_http_server_factory.rs` ?

Comment (bnjjj) [reply apollo-router/src/lib.rs:50]: Indeed :D

**Review by Geal — APPROVED** [review]

**Pull request merged**